cmd/main.go

```diff
@@ -110,34 +108,43 @@ func main() {
 		kubectl)
 	// Add syncer runner
 	if err = mgr.Add(LeaderElectionRunner(syncer.Start)); err != nil {
-		logger.Fatalw("Unable to add autoscaling runner", "err", err)
+		logger.Fatalw("Unable to add autoscaling runner", zap.Error(err))
 	}
-	reconciler, err := controller.NewGitSyncReconciler(
+	gitSyncReconciler, err := controller.NewGitSyncReconciler(
 		mgr.GetClient(),
 		mgr.GetScheme(),
 		configManager,
 		syncer,
 	)
 	if err != nil {
-		logger.Fatalw("Unable to create GitSync controller", err)
+		logger.Fatalw("Unable to create GitSync controller", zap.Error(err))
 	}
 
-	if err = reconciler.SetupWithManager(mgr); err != nil {
-		logger.Fatalw("Unable to set up GitSync controller", err)
+	gitSyncController, err := runtimecontroller.New(ControllerGitSync, mgr, runtimecontroller.Options{
+		Reconciler: gitSyncReconciler,
+	})
+	if err != nil {
+		logger.Fatalw("Unable to set GitSync controller", zap.Error(err))
+	}
 
+	// Watch GitSync objects
+	if err := gitSyncController.Watch(&source.Kind{Type: &apiv1.GitSync{}}, &handler.EnqueueRequestForObject{},
+		predicate.Or(
+			predicate.GenerationChangedPredicate{}, predicate.LabelChangedPredicate{},
+		)); err != nil {
+		logger.Fatalw("Unable to watch GitSyncs", zap.Error(err))
```

Comment:
Interesting. I'd actually added the GenerationChanged Predicate in [this](https://github.com/numaproj-labs/numaplane/pull/54/files) PR but I guess it was removed at some point.

Reply:
Do you have any type of Label specifically in mind as far as watching for Label changes as well?

Reply:
Actually, regarding my first comment, I noticed the code is still in there: `reconciler.SetupWithManager()` does actually look for the `GenerationChangedPredicate{}` still

Reply:
Reverted!